ci-operator/config/openshift-knative/serverless-operator/openshift-knative-serverless-operator-main__ocp4.17-lp-interop.yaml

```diff
@@ -33,9 +33,9 @@ build_root:
 images:
 - dockerfile_path: knative-operator/Dockerfile
   inputs:
-    openshift_release_rhel-8-release-golang-1.22-openshift-4.17:
+    openshift_release_rhel-9-release-golang-1.22-openshift-4.17:
```

Comment:
Is rhel-9 version of the Golang builder required? This looks like a "manual" change in this file but the target Dockerfile still uses rhel-8: https://github.com/openshift-knative/serverless-operator/blob/main/knative-operator/Dockerfile
I think the registry-replacer tool will change this back to rhel-8 whenver it runs next time. The registry-replacer automatically updates CI configs based on existing dockerfiles, one example [here](https://github.com/openshift/release/commit/27fe619e38615fe641b0f485fbe4a34cfde5c556) . This change was not for serverless, I'm just giving an example of how the registry-replacer works.

Reply:
@mgencur - Followed the FIPS checklist from [Enable and Productize FIPS on your Operator](https://docs.google.com/document/d/1CTpSwITQfOgoOTlPITZNJZy0ltYz60hxkmWsLemCqw0/edit#heading=h.xyd5bvyfbwmm) that specified using `registry.ci.openshift.org/ocp/builder:rhel-9-golang-1.22-openshift-4.17` in upstream source repo (eg. [base image](https://github.com/openshift/release/blob/master/ci-operator/config/openshift/cert-manager-operator/openshift-cert-manager-operator-master.yaml#L2-L6)).

This change was tested and passed in both rehearsal jobs for FIPS-enabled and non-FIPS periodic jobs.

Reply:
@dfrazzette I have read through the documents.

The version of Golang builder should be aligned with the version of the base images. We currently use UBI8 as the base image. Our product is staying with UBI8 for now until more certifications are done for RHEL9 (see [link](https://access.redhat.com/articles/compliance_activities_and_gov_standards#fips-140-2-and-fips-140-3-2)) and also because we still support OCP 4.13. The UBI8 image is certified for FIPS (see email "FIPS 140 support for OpenShift bundled products and operators by 2025.01.31 (not Q3 2024)" on  aos-devel from May).

For this reason, we should stay with the RHEL8 version of the Golang builder (to be aligned with the base image).

@Kaustubh-pande (our productization lead) please correct me if I'm wrong with our FIPS compliance and UBI8.

Reply:
Side note: The document also speaks about "fips-check-image-scan" step that could be added to the CI pipeline. It would be really nice to add it here when we expect the test suites to pass for FIPS.

Reply:
Your summary is accurate. We should indeed stay with the RHEL8 version of the Golang builder to maintain alignment with the UBI8 base image.